main.cpp

```diff
@@ -0,0 +1,59 @@
+#include <iostream>
+#include <random>
+#include <regex>
+
+void roll(int numberDice, int diceSides);
+void parseString(char *argument);
+
+int main(int argc, char *argv[]){
+//TODO Add some basic error checking
+    if(argc <= 1){
+        std::cout << "You haven't entered any arguments" << std::endl;
+        return -1;
+    }
+
+    for(int index = 1; index < argc; index++)
+         parseString(argv[index]);
+    return 0;
+}
+
+//Parses input then calls roll() on the arguments
+void parseString(char *argument){
```

Comment:
Passing a char* is okay here, because you're tightly controlling the inputs, but it would be better practice to take a std::string (best would be a const std::string&). Passing a char* to a function in C++ is generally frowned upon, because the length is not known. The call to regex_search below is probably calling strlen on the char* to iterate until it finds a NULL character (signifying the end of a string), so that's how it's working. But in the real world, someone might accidentally call this function with a char* that isn't a properly NULL-terminated string, and then things might get weird.